core/corehttp/gateway_handler.go

```diff
@@ -286,6 +286,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	w.Header().Set("Content-Type", "text/html")
```

Comment:
Let's move this down to where we actually render the template (https://github.com/ipfs/go-ipfs/pull/7330/files#diff-19ceb36d78bde310c54cc44a95bec44cR353).

Reply:
It was done on purpose, ensuring that a successful response to HEAD and GET requests both return the same value. There seem to be some `internalWebError` error cases though where text/html might not be desirable though. I'll have a look.

Reply:
Yeah, I was concerned about the error cases. Maybe just apply it before HEAD?

Reply:
What about explicitly setting `Content-Type: text/plain` if we know for a fact that we are returning a plaintext error message (if this is not already done)? That should treat all cases where this issue could manifest, namely that content-sniffing is prone to emitting the wrong result.

Reply:
Good point. We can set that inside `internalWebError` which should cover all cases. Could you add that change?

Reply:
It looks like error cases are already correctly treated, `http.Error` will overwrite the `Content-Type`.